File: task-03/Ruby/nPrime.rb
def is_prime(num)
  return false if num < 2
  (2..Math.sqrt(num)).each do |i|
    return false if num % i == 0
  end
  true
end

def find_primes(n)
  primes = []
  (2..n).each do |i|
    primes << i if is_prime(i)
  end
  primes
end

puts "Enter a number: "
n = gets.chomp.to_i
result = find_primes(n)
puts "Prime numbers up to #{n} are: #{result}"

File: task-03/Haskell/nPrime.hs
isPrime :: Int -> Bool
isPrime num
    | num < 2 = False
    | otherwise = all (\x -> num `mod` x /= 0) [2..truncate $ sqrt $ fromIntegral num]

findPrimes :: Int -> [Int]
findPrimes n = filter isPrime [2..n]

main :: IO ()
main = do
    putStrLn "Enter a number: "
    n <- readLn
    let result = findPrimes n
    putStrLn $ "Prime numbers up to " ++ show n ++ " are: " ++ show result
